pkg/kube-metrics/collector.go

```diff
@@ -0,0 +1,80 @@
+// Copyright 2019 The Operator-SDK Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package kubemetrics
+
+import (
+	"context"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/watch"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/client-go/tools/cache"
+	kcollector "k8s.io/kube-state-metrics/pkg/collector"
+	"k8s.io/kube-state-metrics/pkg/metric"
+	metricsstore "k8s.io/kube-state-metrics/pkg/metrics_store"
+)
+
+// NewCollectors returns collections of metrics in the namespaces provided, per the api/kind resource.
+// The metrics are registered in the custom generateStore function that needs to be defined.
+func NewCollectors(dclient dynamic.NamespaceableResourceInterface, namespaces []string, api string, kind string, metricFamily []metric.FamilyGenerator) []kcollector.Collector {
```

Comment:
```suggestion
func NewCollectors(dclient dynamic.NamespaceableResourceInterface, namespaces []string, api, kind string, metricFamily []metric.FamilyGenerator) []kcollector.Collector {
```
Just to shorten the function signature a bit. Not necessary.